diff --git a/node/flatpak-node-generator.py b/node/flatpak-node-generator.py
index 63bafb1..74d6c84 100755
--- a/node/flatpak-node-generator.py
+++ b/node/flatpak-node-generator.py
@@ -1291,6 +1291,10 @@ class NpmModuleProvider(ModuleProvider):
 
     async def generate_package(self, package: Package) -> None:
         self.all_lockfiles.add(package.lockfile)
+
+        if package.version.startswith('file:'):
+            return
+
         source = package.source
 
         assert not isinstance(source, ResolvedSource)